### DISCUSSION TITLE
Mainboard/part cooling fans reversed on 4.2.7 board?

### DISCUSSION THREAD

**Discussion by shaungrady** [Jyers/Marlin#318] [Q&A] [answered]
Today I replaced my faulty 4.2.2 board with a 4.2.7 board, and flashed the `v1.2.2` firmware on it, but the fan for the mainboard and the part cooling fan are reversed. The way the two fan connectors are keyed, you can't reverse them on the board itself; the part cooling fan is `K-FAN1` and the board is `K-FAN2`.

I don't know enough about Marlin to say where this issue exists, but it was working correctly on my 4.2.2 board.

**Comment by Jyers** [comment, marked as answer]
That is very strange, if this is the case it would be an issue with the pin definitions for the 4.2.7 board in marlin. I would recommend opening an issue on the upstream marlin branch, as that would be the best place to get such a thing fixed.

**Reply by shaungrady** [reply]
Thanks Jyers. I might have been a bit hasty; it seems setting the fan speed under the **Control** menu affects both the part and board fans. I set it to `255` and tried swapping the `K-FAN` connectors and both were at full throttle. I installed Creality's 4.2.7 firmware and it's the same way.

**Comment by nullstalgia** [comment]
@Jyers Actually, I'm currently having this issue with a 4.2.2 board. Amazon Ender 3 V2 ordered Late July, 2020.

**Reply by nullstalgia** [reply]
Wait, have I fallen for the same trip as Shaun here with the fan speed? :P 

Let me open up my printer and check my connections.

Edit: Swapping the fans also didn't do anything for me. Oh well.

**Comment by DragRedSim** [comment]
This is a design flaw with the E3V2, on both v4.2.2 and v4.2.7 boards. Both `KFAN` connectors are connected to the same input, meaning they cannot be controlled independently. Whatever you set for the part cooling fan, that will also be applied to the motherboard. This means that the motherboard will not be cooled unless the printer is actively running, and above the first layer (dependent on slicer settings). It is likely a good idea to strip the connector of the motherboard fan, and connect it to the fan screw terminals along with the hotend fan; this way, it will always have air running to the motherboard.

**Reply by Cal-LF** [reply]
This has been an issue back to the Ender 3. First mod I did on the Ender 3 and the E3V2 was move the MB fan wires to the hot end terminals.

**Reply by AngleV** [reply]
Me too. And i took it a step further by adding potensiometers and voltage meters both to the mobo fan and the cooling fan.
I know its somehow useless but with this way i can drop the voltage to both of them and reduce the noise they do.
Also it looks nice imho.


![178235273_224200746131294_283334932670199849_n](https://user-images.githubusercontent.com/7430985/115968316-fb259900-a53f-11eb-81e6-68c32c90b5e0.jpg)
![177292443_504918557202256_8148825543909597214_n](https://user-images.githubusercontent.com/7430985/115968319-0082e380-a540-11eb-9ab0-be285108efbe.jpg)
![178539985_772026870175142_6460516455097168157_n](https://user-images.githubusercontent.com/7430985/115968323-04166a80-a540-11eb-816b-ab62c384614e.jpg)
